src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.InfoCache.cs

```diff
@@ -69,6 +69,42 @@ public bool TryGetMemberInfo(IMethodSymbol method, out MemberInfo memberInfo)
                 return memberInfo.LengthLikeProperty != null;
             }
 
+            public bool TryGetMemberInfoOneArgument(IMethodSymbol method, out MemberInfo memberInfo)
+            {
+                if (!IsSliceLikeMethodWithOneArgument(method))
+                {
+                    memberInfo = default;
+                    return false;
+                }
+
+                if (_methodToMemberInfo.TryGetValue(method, out memberInfo))
```

Comment:
```suggestion
                if (!_methodToMemberInfo.TryGetValue(method, out memberInfo))
```

then have hte logic to try to produce the value in the if-statement.

Reply:
Fixed.